### DISCUSSION TITLE
gen_vx_mask wtih shape files?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [dtcenter/METplus#1312] [Configuration] [answered]
Following up on the tutorial session, I don't see an example of using a shape file to generate a mask in the online material. Could someone post a sample command here?

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. I wanted to follow up with this today since it's still fresh from the tutorial. To read in a shapefile, you would use a command similar to this:

`gen_vx_mask ${GRID_FILE} ${SHP_FILE}.shp ${MASK_NAME}.nc -type shape -shapeno ${REC} -name ${VAR_NAME} -v 3`

Stepping through this command, the ${GRID_FILE} is the grid that your ${SHP_FILE}.shp will be mapped to. ${MASK_NAME}.nc is the filename that you'll write the resulting grid out to, and using the `-type shape` flag allows all of this to be possible. the `-shapeno` flag will be used to access the ${SHP_FILE}.shp and extract the ${REC} record for mask creation (make sure you know which entry in the shapefile you're looking for!). The `-name` flag is optional, and sets the mask name inside the ${MASK_NAME}.nc file as ${VAR_NAME}. The `-v` flag is also optional and will control the verbosity of the output for the tool.

I hope that helps you for an example!

**Reply by benjamin-cash** [reply]
Perfect, thanks!
